android-interop-testing/app/src/androidTest/java/io/grpc/android/integrationtest/TesterActivityTest.java

```diff
@@ -0,0 +1,58 @@
+package io.grpc.android.integrationtest;
+
+import android.os.Bundle;
+import android.test.ActivityInstrumentationTestCase2;
+import android.test.InstrumentationTestRunner;
+import junit.framework.Assert;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The Activity test of our App, used to trigger the integration test from commandline.
+ */
+public class TesterActivityTest extends ActivityInstrumentationTestCase2<TesterActivity> {
+
+  private String host;
+  private int port = 8030;
+  private String result;
+  private String serverHostOverride;
+  private String testCase;
+  private boolean useTls;
+  private boolean useTestCa;
+
+  public TesterActivityTest() {
+    super(TesterActivity.class);
+  }
+
+  public void setUp() throws Exception {
+    super.setUp();
+    getActivity();
+    Bundle args = ((InstrumentationTestRunner) getInstrumentation()).getArguments();
+    host = args.getString("server_host", "");
+    port = Integer.parseInt(args.getString("server_port", "0"));
+    assertTrue("Please specify server_host and server_port.", !host.isEmpty() && port != 0);
+    serverHostOverride = args.getString("server_host_override", null);
+    testCase = args.getString("test_case", "empty_unary");
+    useTls = Boolean.parseBoolean(args.getString("use_tls", "true"));
+    useTestCa = Boolean.parseBoolean(args.getString("use_test_ca", "false"));
+  }
+
+  public void testGrpc() throws Exception {
+    final CountDownLatch finished = new CountDownLatch(1);
+
+    new GrpcTestTask(testCase, host, port, serverHostOverride, useTls, useTestCa,
```

Comment:
1. google-auth-library-java would likely use GoogleAuthUtil, but would convert it into a Credential which then we use within an interceptor. It doesn't support it today, but it should later.
 2. [In](https://github.com/grpc/grpc-java/blob/4d5caf83fba1515cd3aa6989578fe77da4cde2af/interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java#L247) [our](https://github.com/grpc/grpc-java/blob/9a6ded523aa9e1c6e03410596d1ade62f903a0fc/netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java) [normal](https://github.com/grpc/grpc-java/blob/4ee2a6584a2f3a2054654409355b1f7995433b96/interop-testing/src/main/java/io/grpc/testing/integration/TestServiceServer.java) [tests](https://github.com/grpc/grpc-java/blob/eb15eb28216fae8dc8cf50e50ad6943ce2234c95/interop-testing/src/test/java/io/grpc/testing/integration/Http2OkHttpTest.java), we use loadCert. That is what [loadCert does](https://github.com/grpc/grpc-java/blob/master/testing/src/main/java/io/grpc/testing/TestUtils.java#L204).